python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -391,18 +429,47 @@ def run(self, inputWarps, skyInfo, **kwargs):
                 warp.mask.array = artifactMask.array
                 del artifactMask
 
-            # Pre-process the warp before coadding.
             # Each Warp that goes into a coadd will typically have an
             # independent photometric zero-point. Therefore, we must scale each
             # Warp to set it to a common photometric zeropoint.
             if self.config.do_scale_zero_point:
-                self.scale_zero_point.run(exposure=warp, dataRef=warpRef)
+                imageScaler = self.scale_zero_point.run(exposure=warp, dataRef=warpRef).imageScaler
+                zero_point_scale_factor = imageScaler.scale
```

Comment:
Are we certain that this is going to be a scalar `ImageScalar` rather than a `SpatialImageScaler` because we're doing it cell-by-cell and evaluating the `PhotoCalib` only at the centers of cells?  If it's a `SpatialImageScaler` it won't have your new `scale` property, because there won't be a single number, and I don't think we can assume the `PhotoCalib` will be spatially constant.

Reply:
Hmm, good point. This works on the current defaults, but let me look into what the best way to obtain a scalar value would be in all cases.

Reply:
Given the [100+ post thread](https://rubin-obs.slack.com/archives/C07TD4VUJ3X/p1740075497308109) we just had, I am even more certain that we shouldn't worry too much to generalize this to work with `SpatialImageScaler`. That, along with `SpatialScaleZeroPointTask` is used nowhere in our code base, not even in unit tests. I'll wait for @fred3m 's changes to come in before I decide how to proceed here.

Reply:
I completely agree.  I hadn't realized the spatial version was going unused before.